src/components/MessageList.tsx

```diff
@@ -0,0 +1,434 @@
+	 const store = useStore();
+
+	 const deleteMessage = (index: number) => {
+		 // eslint-disable-next-line @typescript-eslint/no-explicit-any
+		 const newArray: any[] = [];
+
+		 const tmpIndicators: Indicator[] = store.deleteIndicator(index);
+		 store
+			 .getCurrentMessagesArray()
+			 .forEach((item: ParsedMessageItem | ActMessageItem, i: number) => {
+				 if (i !== index) {
+					 newArray.push(item);
+				 }
+			 });
+		 store.clearParsedMessages();
+		 newArray.forEach((mess: ParsedMessageItem | ActMessageItem, i) => {
+			 store.addParsedMessage(mess, tmpIndicators[i]);
+		 });
+	 };
+
+	 const setDelay = (delay: number) => {
+		 store.setEditedMessageSendDelay(delay);
+	 };
+
+	 return (
+		 <div className='scrolledBlock'>
+			 <ul>
+				 <li>
+					 {editMessageMode ? (
+						 <div
+							 className={'normalNewMessage'}
+							 onClick={() => {
+								 store.setEditMessageMode(false);
+							 }}>
+							 New Message
+						 </div>
+					 ) : null}
+				 </li>
+
+				 {((messages as ParsedMessageItem[]) || (messages as ActMessageItem[])).map(
+					 (item: ParsedMessageItem | ActMessageItem, index: number) => (
+						 <li key={nanoid()} className='messageItem'>
+							 <MessageItem
+								 index={index}
+								 message={item}
+								 indicators={indicators}
+								 editMessageMode={editMessageMode}
+								 editedMessageIndex={editedMessageIndex}
+								 selectMessage={store.selectMessage}
+								 deleteMessage={deleteMessage}
+								 setDelay={setDelay}
+							 />
+						 </li>
+					 ),
+				 )}
+			 </ul>
+		 </div>
+	 );
+};
+
+const EmbeddedEditor = (props: { schema: string; object: string }) => {
+	 const url = `http://localhost:3000?schema=${props.schema}&${
+		 props.object == null ? '' : `object=${props.object}`
+	 }&editorMode=dictionaryEditor&embedded=true`;
+	 return (
+		 <div className='scrolledBlock'>
+			 <iframe className='embeddedEditor' src={url}></iframe>
+		 </div>
+	 );
+};
+
+const MessageEditArea = ({
+	 editMessageMode,
+	 indicators,
+	 messages,
+	 editedMessageIndex,
+	 eventsPanelArea,
+}: MessageEditAreaProps) => {
+	 const store = useStore();
+	 return (
+		 <div className='messageEditArea'>
+			 {store.selectedSchemaType === 'parsed-message' ? (
+				 <SplitView panelArea={eventsPanelArea} onPanelAreaChange={store.setPanelArea}>
+					 <SplitViewPane>
+						 <MessageList
+							 messages={messages}
+							 indicators={indicators}
+							 editMessageMode={editMessageMode}
+							 editedMessageIndex={editedMessageIndex}
+						 />
+					 </SplitViewPane>
+
+					 <SplitViewPane>
+						 <EmbeddedEditor
+							 schema='schema-qa'
+							 object={store.selectedDictionaryName || ''}
+						 />
+					 </SplitViewPane>
+				 </SplitView>
+			 ) : (
+				 <MessageList
+					 messages={messages}
+					 indicators={indicators}
+					 editMessageMode={editMessageMode}
+					 editedMessageIndex={editedMessageIndex}
+				 />
+			 )}
+		 </div>
+	 );
+};
+
+observer(MessageEditArea);
```

Comment:
observer is a higher order component made so that your Component updates when store observables change.
it does not modify component, it returns component wrapped with observer.
So here it returns components into nowhere